tests/test_metadata.py

```diff
@@ -0,0 +1,705 @@
+"""Tests for metadata in IntervalSet, TsdFrame, and TsGroup"""
+
+from numbers import Number
```

Comment:
General comment for these tests, we should have at least more than 1 case for each operation, trying to catch corner cases like:

threshold for splitting intervals sts larger than any given epoch, smaller than each epoch, inbetween... slice 0 element, 1 element, slice with step =1 step >1... stuff like that. For the rest they look good

Reply:
Basically,my main point is try potential corner cases: add metadata of len=1, len>1, len=0, pass a list, an array etc of all these sizes